### PR TITLE
Implements NullReferenceException fix on method

### LINKED ISSUE #300: NullReferenceException: OneSignal.isValidSuccessFailureDelegate
**Description:**

We start having this issue after upgrade to the latest version. According to Crashlytics, it affects more than 3k users every day

**Environment**

1. What version of the Unity OneSignal SDK are you using?
**2.12.2 Release**
1. Does the issue occur only in iOS, Android, etc?
**Andriod Only**

**Anything else:**

Non-fatal Exception: java.lang.Exception: NullReferenceException : Object reference not set to an instance of an object.
       at OneSignal.isValidSuccessFailureDelegate(OneSignal)
       at OneSignal.onPostNotificationFailed(OneSignal)

### PR DESCRIPTION
**Pull request by rgomezp**
Fixes #300

## 
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-unity-sdk/315)

### CODE REVIEW

File: OneSignalExample/Assets/OneSignal/src/OneSignal.cs [hunk 1]
```diff
@@ -1250,7 +1250,7 @@ public class UnityBuilder {
 
         // Some functions have a 'success' and 'failure' delegates, so this validates nothing is missing the json response
         private bool isValidSuccessFailureDelegate(Dictionary<string, object> jsonObject) {
-            if (!isValidDelegate(jsonObject))
+            if (jsonObject==null || !isValidDelegate(jsonObject) || !jsonObject.ContainsKey("delegate_id"))
```
Comment (jkasten2) [inline OneSignalExample/Assets/OneSignal/src/OneSignal.cs:1253]: `!jsonObject.ContainsKey("delegate_id")` is not needed, `isValidDelegate` already checks for this.

Comment (jkasten2) [reply OneSignalExample/Assets/OneSignal/src/OneSignal.cs:1253]: `jsonObject==null` should have a spaces around the `==`
`jsonObject == null`

**Review by jkasten2 — COMMENTED** [review]
Looking at this again thinking we should find the reason why it is null to begin with. Does this only happen with `onPostNotificationFailed` as the issue states?

Can we add a error log when this is `null`?

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/onesignal/onesignal-unity-sdk/315#-:-MXJrgs3Dea_pl5fBs7h:bsfweqv)* status: all files reviewed, 1 unresolved discussion (waiting on @rgomezp)

**Comment by shefich** [comment]
Still have this issue on the version 2.14.3 installed from unity's package manager.
Tried to fix by changing lines from OneSignal.cs to these ones:
"if (jsonObject==null || !isValidDelegate(jsonObject) || !jsonObject.ContainsKey("delegate_id"))"
but nothing changed maybe because now this file is controlled by package manager.
Changes requested!

**Comment by shefich** [comment]
@rgomezp will you fix the issue?
As I mentioned before, your "fix" doesn't work as it doesn't check for the NULL object inside the "isValidDelegate" function and inside "isValidSuccessFailureDelegate" function.
Everytime I should implement my fix in projects from scratch, because now OneSignal is a part of "Packages" and it's code being auto-updated after each resolution of dependencies or assets refreshing.

**Comment by jkasten2** [comment]
Closing this PR as the code has been rewritten in the 3.0.0 release and is no longer an issue.

**Pull request closed without merging**